akka-remote/src/main/scala/akka/remote/artery/LruBoundedCache.scala

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.util.{ OptionVal, Unsafe }
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+
+object FastHash {
+
+  val Prime1: Int = 32717
+  val Prime2: Int = 32749
+  val Prime3: Int = 65519
+  val Prime4: Int = 65521
+
+  // FIXME: !!! Totally unscientific hash !!!, until I figure out how to port xxHash efficiently (it needs 64 bit access to be
+  // fast so we need the ByteBuffer instead of the String)
+  def ofString(s: String): Int = {
```

Comment:
is `String.hashCode` too slow or too weak? One advantage of it performance wise is that it is caching the value.